### PR TITLE
Update Changelog for Release v.4.3.0

### PR DESCRIPTION
**Pull request by lumagi**
@zariiii9003 I tried to remove bugfixes already included in v4.2.2. Please also check if you agree with the sorting and categorization of entries.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,54 @@
...
+### Interface Improvements
+* Change Bus constructor implementation and typing (#1557)
+* Add `protocol` property to BusABC to determine active CAN Protocol (#1532)
+* Add auto-modifying cyclic tasks (#703)
+* NeoVi: Enable send and receive on network ID above 255 (#1627)
+* Optimize PCAN send performance (#1640)
+* Support version string of older PCAN basic API (#1644)
+* Kvaser: add parameter exclusive and `override_exclusive` (#1660)
+* Relax BitTiming & BitTimingFd Validation (#1618)
+* Send HighPriority Message to flush VectorBus Tx buffer (#1636)
```
Comment (zariiii9003) [inline CHANGELOG.md:33]:
```suggestion
* Vector: Send HighPriority Message to flush Tx buffer (#1636)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,54 @@
...
+* PCAN: remove Windows registry check (#1672)
+* Vector: Skip the `can_op_mode check` if the device reports `can_op_mode=0` (#1678)
```
Comment (zariiii9003) [inline CHANGELOG.md:35]: I'd move this to bug fixes.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,54 @@
...
+* Change Bus constructor implementation and typing (#1557)
+* Add `protocol` property to BusABC to determine active CAN Protocol (#1532)
+* Add auto-modifying cyclic tasks (#703)
+* NeoVi: Enable send and receive on network ID above 255 (#1627)
+* Optimize PCAN send performance (#1640)
+* Support version string of older PCAN basic API (#1644)
+* Kvaser: add parameter exclusive and `override_exclusive` (#1660)
+* Relax BitTiming & BitTimingFd Validation (#1618)
+* Send HighPriority Message to flush VectorBus Tx buffer (#1636)
+* PCAN: remove Windows registry check (#1672)
```
Comment (zariiii9003) [inline CHANGELOG.md:34]:
```suggestion
* PCAN: remove Windows registry check to fix 32bit compatibility (#1672)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,54 @@
...
+* NeoVi: Fixed serial number range (#1650)
+* Fix PCAN timestamp (#1651)
+* Fix BLFReader error for incomplete or truncated stream (#1662)
+* We do not need to account for drift when we `USE_WINDOWS_EVENTS` (#1666)
+
+Features
+--------
+
+### IO
+* Distinguish Text/Binary-IO for Reader/Writer classes. (#1585)
```
Comment (zariiii9003) [inline CHANGELOG.md:21]: This is just a type annotation issue, i'd move this to miscellaneous

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,54 @@
...
+* Can Player compatibility with interfaces that use additional configuration (#1610)
+
+### Interface Improvements
+* Change Bus constructor implementation and typing (#1557)
+* Add `protocol` property to BusABC to determine active CAN Protocol (#1532)
+* Add auto-modifying cyclic tasks (#703)
```
Comment (zariiii9003) [inline CHANGELOG.md:27]:
```suggestion
* Add `modifier_callback` parameter to `BusABC.send_periodic` for auto-modifying cyclic tasks (#703)
```

Add an "API" sub section to "Features", there are a few points here, which i'd move to API

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,54 @@
...
+* NeoVi: Enable send and receive on network ID above 255 (#1627)
+* Optimize PCAN send performance (#1640)
+* Support version string of older PCAN basic API (#1644)
+* Kvaser: add parameter exclusive and `override_exclusive` (#1660)
+* Relax BitTiming & BitTimingFd Validation (#1618)
```
Comment (zariiii9003) [inline CHANGELOG.md:32]:
```suggestion
* Add optional `strict` parameter to relax BitTiming & BitTimingFd Validation (#1618)
```
Move to API

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,54 @@
...
+* Vector: Skip the `can_op_mode check` if the device reports `can_op_mode=0` (#1678)
+
+### Miscellaneous
+* Convert setup.py to pyproject.toml (#1592)
+* activate ruff pycodestyle checks (#1602)
+* Update linter instructions in development.rst (#1603)
+* remove unnecessary script files (#1604)
+* BigEndian test fixes (#1625)
+* Use same configuration file as Linux on macOS (#1657)
+* Catch `pywintypes.error` in broadcast manager (#1659)
```
Comment (zariiii9003) [inline CHANGELOG.md:44]: Move to bug fixes

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,54 @@
...
+* Update linters, activate more ruff rules (#1669)
+* align `ID:` in can.Message string (#1635)
+* Do not stop notifier if exception was handled (#1645)
```
Comment (zariiii9003) [inline CHANGELOG.md:47]: This one could be a breaking change 🤔

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,54 @@
...
+* Add `BitTiming.iterate_from_sample_point` static methods (#1671)
```
Comment (zariiii9003) [inline CHANGELOG.md:48]: Move to API

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,54 @@
...
+* Distinguish Text/Binary-IO for Reader/Writer classes. (#1585)
+* Can Player compatibility with interfaces that use additional configuration (#1610)
+
+### Interface Improvements
+* Change Bus constructor implementation and typing (#1557)
+* Add `protocol` property to BusABC to determine active CAN Protocol (#1532)
+* Add auto-modifying cyclic tasks (#703)
+* NeoVi: Enable send and receive on network ID above 255 (#1627)
+* Optimize PCAN send performance (#1640)
+* Support version string of older PCAN basic API (#1644)
```
Comment (zariiii9003) [inline CHANGELOG.md:30]:
```suggestion
* PCAN: Support version string of older PCAN basic API (#1644)
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,54 @@
...
+### IO
+* Distinguish Text/Binary-IO for Reader/Writer classes. (#1585)
+* Can Player compatibility with interfaces that use additional configuration (#1610)
+
+### Interface Improvements
+* Change Bus constructor implementation and typing (#1557)
+* Add `protocol` property to BusABC to determine active CAN Protocol (#1532)
+* Add auto-modifying cyclic tasks (#703)
+* NeoVi: Enable send and receive on network ID above 255 (#1627)
+* Optimize PCAN send performance (#1640)
```
Comment (zariiii9003) [inline CHANGELOG.md:29]:
```suggestion
* PCAN: Optimize send performance (#1640)
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,54 @@
...
+Features
+--------
+
+### IO
+* Distinguish Text/Binary-IO for Reader/Writer classes. (#1585)
+* Can Player compatibility with interfaces that use additional configuration (#1610)
+
+### Interface Improvements
+* Change Bus constructor implementation and typing (#1557)
+* Add `protocol` property to BusABC to determine active CAN Protocol (#1532)
```
Comment (zariiii9003) [inline CHANGELOG.md:26]: Move to API

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,3 +1,54 @@
+Version 4.3.0
+===========
+
+Breaking Changes
+----------------
+* Raise Minimum Python Version to 3.8 (#1597)
+
+Bug Fixes
+---------
+* Fix Vector channel detection (#1634)
```
Comment (zariiii9003) [inline CHANGELOG.md:10]:
```suggestion
* Vector: channel detection fails, if there is an active flexray channel  (#1634)
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,3 +1,54 @@
...
+* ixxat: Fix exception in 'state' property on bus coupling errors (#1647)
+* NeoVi: Fixed serial number range (#1650)
+* Fix PCAN timestamp (#1651)
```
Comment (zariiii9003) [inline CHANGELOG.md:13]:
```suggestion
* PCAN: Fix timestamp offset due to timezone (#1651)
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,3 +1,54 @@
...
+* Fix BLFReader error for incomplete or truncated stream (#1662)
+* We do not need to account for drift when we `USE_WINDOWS_EVENTS` (#1666)
```
Comment (zariiii9003) [inline CHANGELOG.md:15]: Move to miscellaneous, it's not really a bug.

**Review by zariiii9003 — CHANGES_REQUESTED** [review]

**Comment by lumagi** [comment]
Sorry for the many commits, I probably could have batched your change request.

**Comment by zariiii9003** [comment]
You missed a few comments 😄

File: CHANGELOG.md (hunk 2 above)
Comment (lumagi) [reply CHANGELOG.md]: You mean line 35 only, right?

File: CHANGELOG.md (hunk 6 above)
Comment (lumagi) [reply CHANGELOG.md]: I think this one got lost.

File: CHANGELOG.md (hunk 7 above)
Comment (lumagi) [reply CHANGELOG.md:48]: And this one

**Comment by lumagi** [comment]
You're right, I think Github was trying to be nice by compacting the view there.

File: CHANGELOG.md (hunk 2 above)
Comment (zariiii9003) [reply CHANGELOG.md]: Yes, line 35

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,3 +1,57 @@
...
+### Miscellaneous
+* Distinguish Text/Binary-IO for Reader/Writer classes. (#1585)
+* Convert setup.py to pyproject.toml (#1592)
+* activate ruff pycodestyle checks (#1602)
+* Update linter instructions in development.rst (#1603)
+* remove unnecessary script files (#1604)
+* BigEndian test fixes (#1625)
+* align `ID:` in can.Message string (#1635)
+* Use same configuration file as Linux on macOS (#1657)
+* We do not need to account for drift when we `USE_WINDOWS_EVENTS` (#1666)
```
Comment (zariiii9003) [inline CHANGELOG.md:50]:
```suggestion
* We do not need to account for drift when we `USE_WINDOWS_EVENTS` (#1666, #1679)
```

**Comment by zariiii9003** [comment]
@lumagi Can you add #1607 and bump the version to `4.3.0rc0`?

**Review by zariiii9003 — APPROVED** [review]

**Comment by zariiii9003** [comment]
@lumagi It's released: https://pypi.org/project/python-can/4.3.0rc0/
Thank you! 😃

**Comment by lumagi** [comment]
Great, thanks for pushing the release!

**Comment by zariiii9003** [comment]
@hardbyte Would you like to release 4.3? Two weeks passed and so far no complaints about the pre-release...

**Comment by hardbyte** [comment]
Let's do it!

**Comment by lumagi** [comment]
> @hardbyte Would you like to release 4.3? Two weeks passed and so far no complaints about the pre-release...

Can I support here?

**Comment by hardbyte** [comment]
That would be great thanks - I'll open a PR bumping the version but feel free to check and edit the changelog.

**Pull request merged**